opm/parser/eclipse/Parser/Parser.cpp

```diff
@@ -42,60 +42,194 @@
 #include <opm/parser/eclipse/RawDeck/RawConsts.hpp>
 #include <opm/parser/eclipse/RawDeck/RawEnums.hpp>
 #include <opm/parser/eclipse/RawDeck/RawKeyword.hpp>
+#include <opm/parser/eclipse/Utility/Stringview.hpp>
 
 namespace Opm {
 
+    template< typename Itr >
+    static inline Itr find_comment( Itr begin, Itr end ) {
+
+        auto itr = std::find( begin, end, '-' );
+        for( ; itr != end; itr = std::find( itr + 1, end, '-' ) )
+            if( (itr + 1) != end &&  *( itr + 1 ) == '-' ) return itr;
+
+        return end;
+    }
+
+    template< typename Itr, typename Term >
+    static inline Itr strip_after( Itr begin, Itr end, Term terminator ) {
+
+        auto pos = terminator( begin, end );
+
+        if( pos == end ) return end;
+
+        auto qbegin = std::find_if( begin, end, RawConsts::is_quote );
+
+        if( qbegin == end || qbegin > pos )
+            return pos;
+
+        auto qend = std::find( qbegin + 1, end, *qbegin );
+
+        // Quotes are not balanced - probably an error?!
+        if( qend == end ) return end;
+
+        return strip_after( qend + 1, end, terminator );
+    }
+
+    /**
+       This function will remove return a copy of the input string
+       where all characters following '--' are removed. The function
+       handles quoting with single quotes and double quotes:
+
+         ABC --Comment                =>  ABC
+         ABC '--Comment1' --Comment2  =>  ABC '--Comment1'
+         ABC "-- Not balanced quote?  =>  ABC "-- Not balanced quote?
+    */
+    static inline string_view strip_comments( string_view str ) {
+        return { str.begin(), strip_after( str.begin(), str.end(),
+                find_comment< string_view::const_iterator > ) };
+    }
+
+    /* stripComments only exists so that the unit tests can verify it.
+     * strip_comment is the actual (internal) implementation
+     */
+    std::string Parser::stripComments( const std::string& str ) {
+        return { str.begin(), strip_after( str.begin(), str.end(),
+                find_comment< std::string::const_iterator > ) };
+    }
+
+    template< typename Itr >
+    static inline Itr trim_left( Itr begin, Itr end ) {
+
+        return std::find_if_not( begin, end, RawConsts::is_separator );
+    }
+
+    template< typename Itr >
+    static inline Itr trim_right( Itr begin, Itr end ) {
+
+        std::reverse_iterator< Itr > rbegin( end );
+        std::reverse_iterator< Itr > rend( begin );
+
+        return std::find_if_not( rbegin, rend, RawConsts::is_separator ).base();
+    }
+
+    static inline string_view trim( string_view str ) {
+        auto fst = trim_left( str.begin(), str.end() );
+        auto lst = trim_right( fst, str.end() );
+        return { fst, lst };
+    }
+
+    static inline string_view strip_slash( string_view view ) {
```

Comment:
It is, I'm happy to rename it. It was named this way to be consistent with strip_comments.